packages/react-components/src/components/TextFieldWithMention/mentionTagHelpers.ts

```diff
@@ -0,0 +1,957 @@
+// Copyright (c) Microsoft Corporation.
```

Comment:
These code are more like logics for handling html + tags, I wouldn't be too worried about overall structure here, but would be nice to get unit tests for some important functions so no one will mess them up in the future

Not for this PR though, could be another PR for some new work item

Reply:
> These code are more like logics for handling html + tags, I wouldn't be too worried about overall structure here, but would be nice to get unit tests for some important functions so no one will mess them up in the future
> 
> Not for this PR though, could be another PR for some new work item

We do have separated tickets for adding unit tests for all these functions.